### PR TITLE
ZMS-460: Additional notes in ephemeraldata.adoc

### PR DESCRIPTION
**Pull request by ilyusha**
Added a section "Additional Notes on Ephemeral Data" that covers several topics, primarily regarding migration.

### CODE REVIEW

File: ephemeraldata.adoc [hunk 1]
```diff
@@ -58,6 +58,35 @@ zmprov mcf zimbraEphemeralBackendURL ssdb:<ip address|hostname>:port # substitut
 
 ----
 sudo su - zimbra
-# Specify -r for a dry run
 /opt/zimbra/bin/zmmigrateattrs zimbraAuthTokens zimbraCsrfTokenData zimbraLastLogonTimestamp -n 4
```
Comment (fatkudu) [inline ephemeraldata.adoc]: Remove the explicit names of attributes now they aren't needed?

File: ephemeraldata.adoc [hunk 2]
```diff
@@ -58,6 +58,35 @@ zmprov mcf zimbraEphemeralBackendURL ssdb:<ip address|hostname>:port # substitut
...
 ----
+
+== Additional Notes on Ephemeral Data
+
+=== Advanced Migration Options
+
+The `zmmigrateattrs` tool provides several migration options, to be used with careful consideration:
+- By default, ephemeral data is deleted from LDAP as soon as it is migrated to `SSDB`. The `-k` or `--keep-old` option will cause this data to remain in LDAP. This makes it possible to revert the ephemeral backend back to LDAP immediately after migration with no data loss. However, this data will otherwise remain in LDAP and quickly become obsolete.
```
Comment (fatkudu) [inline ephemeraldata.adoc]: Add a line here before this, otherwise this isn't the first list item.

File: ephemeraldata.adoc [hunk 3]
```diff
@@ -58,6 +58,35 @@ zmprov mcf zimbraEphemeralBackendURL ssdb:<ip address|hostname>:port # substitut
...
+
+=== Changes to zmprov
+
+Due to changes in the way multi-valued ephemeral data is stored, the attributes `zimbraAuthTokens` and `zimbraCsrfTokenData` are no longer returned as part of the `zmprov ga <account>` response. The value of `zimbraLastLogonTimestamp` is returned as before.
+
+=== Migration In-Progress Flag
+
+The `zmmigrateattrs` script sets a flag in `SSDB` at the beginning of the migration process and unsets it when migration completes. When this flag is set, the server will look for an account's ephemeral data in both the `SSDB` and LDAP backends, as the data may reside in either one. In the event of an error during migration, the flag will remain set until migration is re-ran successfully. For testing or debugging purposes, this flag can be set or unset via command line using the `--set-flag` and `--unset-flag` options of `zmmigrateattrs`.
```
Comment (fatkudu) [inline ephemeraldata.adoc]: s/re-ran/re-run/

File: ephemeraldata.adoc [hunk 4]
```diff
@@ -58,6 +58,35 @@ zmprov mcf zimbraEphemeralBackendURL ssdb:<ip address|hostname>:port # substitut
...
+
+=== Migration CSV Output
+
+Each run of `zmmigrateattrs` generates a CSV file in `/opt/zimbra/data/tmp/` directory. The file contains migration info for every migrated account. Migration errors are logged in a separate CSV file in the same directory.
```
Comment (fatkudu) [inline ephemeraldata.adoc]: > Migration errors are logged in a separate CSV file in the same directory.

Replace with:

If any migrations fail, a cutdown CSV file report detailing only the errors is also created in the same directory.
The name(s) of the file(s) are logged at the end of the run.

**Comment by fatkudu** [comment]
Looks good apart from minor comments.

**Comment by gordyt** [comment]
+1 Ilya this looks good.  I just generated the guide from your branch and reads through everything.

**Pull request merged**